docs/DeveloperGuide.adoc

```diff
@@ -337,10 +421,146 @@ Use case ends.
 * 3a. The given index is invalid.
 +
 [none]
-** 3a1. AddressBook shows an error message.
+** 3a1. Mortago shows an error message.
 +
 Use case resumes at step 2.
 
+[discrete]
+=== Use case: Find entry
+*MSS*
+
+1.  Mortuary manager switches to the desired view (bodies or workers).
+2.  Mortuary manager specifies word to search.
+3.  Mortago shows a list of entries whose names matches the word.
++
+Use case ends.
+
+*Extensions*
+
+[none]
+* 3a. The list is empty.
++
+Use case ends.
+
+[discrete]
+=== Use case: Filter entries
+*MSS*
+
+1.  Mortuary manage switches to the desired view (bodies or workers).
+2.  Mortuary manager specifies criteria for filter.
+3.  Mortago shows a list of entries that matches the criteria.
++
+Use case ends.
+
+*Extensions*
+
+[none]
+* 3a. The list is empty.
++
+Use case ends.
+
+[discrete]
+=== Use case: Sort entries
+*MSS*
+
+1.  Mortuary manage switches to the desired view (bodies or workers).
+2.  Mortuary manager specifies criteria for sorting.
+3.  Mortago shows a list of entries sorted according to the specified criteria.
++
+Use case ends.
+
+*Extensions*
+
+[none]
+* 3a. The list is empty.
++
+Use case ends.
+
+[discrete]
+=== Use case: Generate report
+
+*MSS*
+
+1.  Mortuary manager requests to generate report for a specific body.
+2.  Mortago creates a new PDF report with body ID as the title.
++
+Use case ends.
+
+*Extensions*
+
+[none]
+* 1a. The given body ID is invalid.
++
+[none]
+** 1a1. Mortago shows an error message.
++
+Use case resumes at step 1.
+
+[discrete]
+=== Use case: Alert for unclaimed bodies.
+
+*MSS*
+
+1.  Mortuary manager wants to be reminded of the next line of action if next of kin cannot be contacted within 24 hours.
+2.  Mortago maintains a record of all the alerts about unidentified and unclaimed bodies until their status is changed.
+3.  Mortago shows pop-up alerts after 24 hours from the point of arrival of the body in the mortuary.
++
+Use case ends.
+
+*Extensions*
+
+[none]
+* 1a. There are no alerts
++
+Use case ends.
+
+
+[discrete]
+=== Use case: Undoing a previous command
+
+*MSS*
+
+1. Mortuary manager requests to undo the previous command.
+2. Mortago undoes the command.
+3. Mortago updates the GUI to reflect the new changes.
+
++
+Use case ends.
+
+*Extensions*
+
+[none]
+* 2a. There is no command to undo.
++
+Use case ends.
+* 2b. An error occurred when undoing the command.
+    ** 2b1. Mortago shows an error message and nothing is changed.
++
+Use case ends.
+
+[discrete]
+=== Use case: Redoing an undone command
+
+*MSS*
+
+1. Mortuary manager requests to redo the last undone command.
+2. Mortago redoes the command.
+3. Mortago updates the GUI to reflect the new changes.
+
++
+Use case ends.
+
+*Extensions*
+
+[none]
+* 2a. There is no command to redo.
++
+Use case ends.
+* 2b. An error occurred when undoing the command.
+    ** 2b1. Mortago shows an error message and nothing is changed.
++
+Use case ends.
+
 _{More to be added}_
 
 [appendix]
```

Comment:
Recommended to add additional Non-Functional Requirements and delete irrelevant ones.